Download plan query from resource URI

br648 opened the pull request:
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing
- [x] The description lists all relevant PRs included in this release _(remove this if not merging to master)_

### Description

Update to download a plan query from https://raw.githubusercontent.com/ibi-group/open-source-configurations/main/atlanta/arc/planQuery.graphql.

br648 committed:
refactor(Updated to load plan query from resource URI): Now downloads the query from a public repo o

binh-dam-ibigroup commented on src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java:
```diff
@@ -1,52 +1,68 @@
      */
-    public static final String PLAN_QUERY_RESOURCE = "queries/planQuery.graphql";
+    private static final String PLAN_QUERY_RESOURCE_URI =
+        getConfigPropertyAsText("PLAN_QUERY_RESOURCE_URI", null);
```
Should there be a default one? Either a default URL or default content?

br648 replied:
So I have updated to include `https://raw.githubusercontent.com/ibi-group/open-source-configurations/main/atlanta/arc/planQuery.graphql` with the assumption any changes will be applied to this. It is worth bearing in mind that if another plan query is used, but is not defined, unexpected results will occur.

binh-dam-ibigroup replied:
No, that custom query template should not be the default.

br648 replied:
@binh-dam-ibigroup do you have the default so I can upload and reference?

binh-dam-ibigroup replied:
@br648 Use the one that OTP-UI/OTP-RR use by default at https://raw.githubusercontent.com/opentripplanner/otp-ui/refs/heads/master/packages/core-utils/src/planQuery.graphql (we might move it in the future but for now it lives there)

br648 replied:
Updated in: https://github.com/ibi-group/otp-middleware/pull/261/commits/695947a38032c6217feaa629bcd7c39ccb538485

JymDyerIBI reviewed (approved):
👍  Looks good to me.

br648 committed:
refactor(GraphQLUtils.java): Now includes a default plan query URI

br648 committed:
refactor(GraphQLUtils.java): Now referencing the correct default plan query resource

binh-dam-ibigroup commented on configurations/default/env.yml.tmp:
```diff
@@ -93,4 +93,7 @@ TRIP_INSTRUCTION_UPCOMING_RADIUS: 10
+US_RIDE_GWINNETT_BUS_OPERATOR_NOTIFIER_QUALIFYING_ROUTES: agency_id:route_id
+
+# The location for an OTP plan query request.
+PLAN_QUERY_RESOURCE_URI: https://plan.resource.com
```
Please add this parameter to the config schema (and README).

br648 replied:
Fixed in: https://github.com/ibi-group/otp-middleware/pull/261/commits/4539fc0123af8deee15cdd5ac87a6acc0c050a72

binh-dam-ibigroup commented on src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java:
```diff
@@ -1,52 +1,71 @@
+    private static final String PLAN_QUERY_RESOURCE_URI =
+        getConfigPropertyAsText(
+            "PLAN_QUERY_RESOURCE_URI",
+            "https://raw.githubusercontent.com/opentripplanner/otp-ui/refs/heads/master/packages/core-utils/src/planQuery.graphql"
```
Extract a constant for this.

br648 replied:
Fixed in: https://github.com/ibi-group/otp-middleware/pull/261/commits/4539fc0123af8deee15cdd5ac87a6acc0c050a72

binh-dam-ibigroup commented on src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java:
```diff
@@ -1,52 +1,71 @@
+    static String getPlanQueryFromResource() {
+        if (GraphQLUtils.PLAN_QUERY_RESOURCE_URI == null) {
+            LOG.error("The plan query resource URI parameter \"PLAN_QUERY_RESOURCE_URI\" is undefined.");
+            return null;
```
Does this apply anymore, now that there is a default URL?

br648 replied:
Fixed in: https://github.com/ibi-group/otp-middleware/pull/261/commits/4539fc0123af8deee15cdd5ac87a6acc0c050a72

binh-dam-ibigroup commented on src/main/java/org/opentripplanner/middleware/utils/GraphQLUtils.java:
```diff
@@ -1,52 +1,71 @@
+    static String planQueryTemplateAsString() {
+        String rawPlanQuery = getPlanQueryFromResource();
+        if (rawPlanQuery == null) {
+            LOG.error("Unable to retrieve plan query from resource: {}.", GraphQLUtils.PLAN_QUERY_RESOURCE_URI);
```
If this method returns null, how is that subsequently handled?

br648 replied:
Fixed in: https://github.com/ibi-group/otp-middleware/pull/261/commits/4539fc0123af8deee15cdd5ac87a6acc0c050a72

binh-dam-ibigroup reviewed (changes requested):
Please add code to pass the query variable `mobilityProfile` to the GraphQL template.

br648 committed:
refactor(Various updates to address PR feedback):

br648 commented:
> Please add code to pass the query variable `mobilityProfile` to the GraphQL template.

@binh-dam-ibigroup I'm not entirely sure what you mean here? I've updated the template to include the `mobilityProfile` param:

https://github.com/ibi-group/open-source-configurations/blob/main/atlanta/arc/planQuery.graphql

I hope this is correct. Lemme know if you were expecting something else!

br648 committed:
Merge branch 'dev' into feature/otp-1392-download-plan-query

binh-dam-ibigroup commented:
> > Please add code to pass the query variable `mobilityProfile` to the GraphQL template.
> 
> @binh-dam-ibigroup I'm not entirely sure what you mean here? I've updated the template to include the `mobilityProfile` param:
> 
> https://github.com/ibi-group/open-source-configurations/blob/main/atlanta/arc/planQuery.graphql
> 
> I hope this is correct. Lemme know if you were expecting something else!

During itinerary checks, the `mobilityProfile` field needs to be populated. As written, it is not (see screenshot):
<img width="449" alt="image" src="https://github.com/user-attachments/assets/63535302-bfc0-4905-a81e-4791aa40a3f7">

binh-dam-ibigroup commented:
> > > Please add code to pass the query variable `mobilityProfile` to the GraphQL template.
> > 
> > 
> > @binh-dam-ibigroup I'm not entirely sure what you mean here? I've updated the template to include the `mobilityProfile` param:
> > https://github.com/ibi-group/open-source-configurations/blob/main/atlanta/arc/planQuery.graphql
> > I hope this is correct. Lemme know if you were expecting something else!
> 
> During itinerary checks, the `mobilityProfile` field needs to be populated. As written, it is not (see screenshot)

To do that, I think you'll have to complete the implementation of `Otp2QueryParams.clone` to include `mobilityProfile`.

br648 committed:
refactor(Addressed PR feedback): Mobility profile now available as part of the itinerary checks

br648 commented:
> > > Please add code to pass the query variable `mobilityProfile` to the GraphQL template.
> > 
> > 
> > @binh-dam-ibigroup I'm not entirely sure what you mean here? I've updated the template to include the `mobilityProfile` param:
> > https://github.com/ibi-group/open-source-configurations/blob/main/atlanta/arc/planQuery.graphql
> > I hope this is correct. Lemme know if you were expecting something else!
> 
> During itinerary checks, the `mobilityProfile` field needs to be populated. As written, it is not (see screenshot): <img alt="image" width="449" src="https://private-user-images.githubusercontent.com/56846598/376733453-63535302-bfc0-4905-a81e-4791aa40a3f7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkwNzMxMDIsIm5iZiI6MTcyOTA3MjgwMiwicGF0aCI6Ii81Njg0NjU5OC8zNzY3MzM0NTMtNjM1MzUzMDItYmZjMC00OTA1LWE4MWUtNDc5MWFhNDBhM2Y3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDE2VDEwMDAwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBhYzZmZGJiYzM4ZmVjM2Y0MjlkZmEyZTdkZTI0M2M5YjM3YWQ1MzQ2OTlmNjc4NThhYzJjOTIyZmY1MDI4NDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.wr3ShM_XdLvRrp-MtdwFlhx_CBoOCsvdlKG6oar5m38">

Fixed in: https://github.com/ibi-group/otp-middleware/pull/261/commits/e203492294f2fc48eeb66197c292d0ff97eb40c7

binh-dam-ibigroup reviewed (approved):
Nice, it works with custom plan query and the default one.
